File: lclang/src/main/java/lclang/statements/expressions/ValueExpression.java
package lclang.statements.expressions;

import lclang.Caller;
import lclang.LCBaseExecutor;
import lclang.Value;
import lclang.exceptions.LCLangRuntimeException;
import lclang.libs.lang.classes.LCClass;

public class ValueExpression extends Expression {
    public final LCClass value;

    public ValueExpression(LCClass value) {
        super(0);
        this.value = value;
    }

    @Override
    public Value visit(Caller prevCaller, LCBaseExecutor visitor) throws LCLangRuntimeException {
        return value.asValue();
    }
}

File: lclang/src/main/kotlin/lclang/statements/ValueExpression.kt
package lclang.statements

import lclang.Caller
import lclang.LCBaseExecutor
import lclang.Value

class ValueExpression(
    val value: Value,
    line: Int,
    column: Int
): Expression(line, column) {
    override fun visit(prevCaller: Caller, visitor: LCBaseExecutor) = value
}